web/api/create-team/index.ts

```diff
@@ -72,197 +81,166 @@ export const POST = withApiAuthRequired(async (req: NextRequest) => {
 
-      return errorResponse({
-        statusCode: 500,
-        code: "Failed to signup",
-        req,
-      });
-    }
+  try {
+    const { insert_team_one } = await getInsertTeamSdk(client).InsertTeam({
+      team_name,
+    });
 
-    user = insertMembershipResult.insert_membership_one.user;
-  } else {
-    let insertTeamResult: InsertTeamMutation | null = null;
+    insertedTeam = insert_team_one;
+  } catch (error) {
+    logger.error("Error while inserting team on create team:", { error });
 
-    try {
-      insertTeamResult = await getInsertTeamSdk(client).InsertTeam({
-        team_name,
-      });
+    return errorResponse({
+      statusCode: 500,
+      code: "Failed to create team",
+      req,
+    });
+  }
 
-      if (!insertTeamResult?.insert_team_one?.id) {
-        throw new Error("Failed to insert team");
-      }
-    } catch (error) {
-      logger.error("Error while inserting team on signup:", { error });
+  // ANCHOR: Insert user
+  let nullifier_hash: string | undefined = undefined;
 
-      return errorResponse({
-        statusCode: 500,
-        code: "Failed to signup",
-        req,
-      });
-    }
+  if (!isEmailUser(auth0User)) {
+    const nullifier = auth0User.sub.split("|")[2];
+    nullifier_hash = nullifier;
+  }
 
-    let insertUserResult: InsertUserMutation | null = null;
+  let insertedUser: InsertUserMutation["insert_user_one"] | null = null;
 
+  if (!hasMemberships) {
     try {
-      insertUserResult = await getInsertUserSdk(client).InsertUser({
+      const { insert_user_one } = await getInsertUserSdk(client).InsertUser({
         user_data: {
-          name: auth0User.name,
-          auth0Id: auth0User.sub,
-          team_id: insertTeamResult.insert_team_one.id,
           ironclad_id: ironCladUserId,
+          auth0Id: auth0User.sub,
+          name: auth0User.name ?? "",
+
           ...(nullifier_hash ? { world_id_nullifier: nullifier_hash } : {}),
 
           ...(auth0User.email_verified && auth0User.email
             ? { email: auth0User.email }
             : {}),
+
+          team_id: insertedTeam?.id,
         },
       });
 
-      if (!insertUserResult?.insert_user_one?.id) {
-        throw new Error("Failed to insert user");
-      }
+      insertedUser = insert_user_one;
     } catch (error) {
-      logger.error("Error while inserting user on signup:", { error });
+      logger.error("Error while inserting user on create team:", { error });
 
       return errorResponse({
         statusCode: 500,
-        code: "Failed to signup",
+        code: "Failed to create team",
         req,
       });
     }
+  }
+
+  // ANCHOR: Insert membership
+  let insertedMembership:
+    | InsertMembershipMutation["insert_membership_one"]
+    | null = null;
+
+  try {
+    const user_id = hasMemberships
+      ? (auth0User as Auth0SessionUser["user"])?.hasura.id
+      : insertedUser?.id;
+
+    if (!insertedTeam?.id) {
+      throw new Error("Team id is null");
```

Comment:
we should be handling these errors and returning right 

```suggestion
      return errorResponse({
      statusCode: 400,
      code: "required",
      detail: " Team ID and User ID are required"
      req,
    });
```